scripts/apps/search/components/item-list-popup.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {ElementType, CSSProperties, SyntheticEvent} from 'react';
+import {createPortal} from 'react-dom';
+import {positionPopup} from '../helpers';
+
+interface IProps {
+    label: string;
+    close: () => void;
+    target: ElementType | EventTarget;
+}
+
+const BACKDROP_STYLE: CSSProperties = {
+    position: 'absolute',
+    top: 0,
+    bottom: 0,
+    left: 0,
+    right: 0,
+    zIndex: 999, // positionPopup sets it to 1000
+    overflow: 'auto', // to allow scroll
+};
+
+export class ItemListPopup extends React.PureComponent<IProps> {
+    constructor(props) {
+        super(props);
+        this.close = this.close.bind(this);
+    }
+
+    componentDidMount() {
+        positionPopup(this.props.target);
+    }
+
+    close(event: SyntheticEvent) {
+        event.stopPropagation();
+        this.props.close();
+    }
+
+    render() {
+        return createPortal(
+            <ul className="highlights-list-menu">
+                <li>
+                    <div className="dropdown__menu-label">{this.props.label}</div>
+                    <button className="dropdown__menu-close" onClick={this.close}>
+                        <i className="icon-close-small icon--white" />
+                    </button>
+                </li>
+                <li>
+                    {this.props.children}
+                </li>
+                {createPortal(
```

Comment:
Why do you need nested portals?

Reply:
that backdrop must be rendered outside react placeholder, it's positioned absolutely so it wouldn't be possible to cover the whole screen if it's inside it

Reply:
What are you using a backdrop for here? It's a popup, not a modal, so it shouldn't make the entire screen darker when it opens. Or is it for something different?

Reply:
to be able to catch scroll or click outside of it